.gitignore

```diff
@@ -40,3 +40,20 @@ objs/
 # Other
 *~
 GeneratedCode
+
+.venv/
+*.pyc
+
+srcCxx/shape.cxx
+srcCxx/shape.h
+srcCxx/shapePlugin.cxx
+srcCxx/shapePlugin.h
+srcCxx/shapeSupport.cxx
+srcCxx/shapeSupport.h
+
+.vscode/c_cpp_properties.json
+.vscode/launch.json
+.vscode/settings.json
+.vscode/tasks.json
+.gitignore
```

Comment:
we need to track the .gitignore, otherwise the changes to this file won't be tracked

Reply:
i don't understand this one